.github/workflows/ci.yml

```diff
@@ -47,14 +47,19 @@ jobs:
     - name: List installed Python packages
       run: python -m pip list
 
-    - name: Test with pytest
+    - name: Test with pytest and coverage
       run: |
-        pytest --ignore tests/benchmarks/ --ignore tests/contrib --ignore tests/test_notebooks.py
+        coverage run --module pytest --ignore tests/contrib --ignore tests/benchmarks --ignore tests/test_notebooks.py
 
     - name: Launch a tmate session if tests fail
       if: failure() && github.event_name == 'workflow_dispatch'
       uses: mxschmitt/action-tmate@v3
 
+    - name: Coverage report for core project
+      run: |
+        coverage report
+        coverage xml
```

Comment:
Need to now generate the report for uploading, but the report is now a clear stage by itself.